### PR TITLE
feat: upgrade mongodb node driver (DEV-4642)

### PR DESCRIPTION
**Pull request by rombat**
BREAKING CHANGE : due to v5 => v6 breaking changes updates:

- `findOneAndX` with implicit `{ includeResultMetadata: false }` (returns value)

unit tests fixed
image

### CODE REVIEW

File: services/auth/lib/local.js [hunk 1]
```diff
@@ -436,6 +437,7 @@ const resetPassword = async function (req, res) {
 const updatePassword = async function (req, res) {
   const missingParams = getMissingParameters(req.body, ['new', 'confirm']);
   const passwordRegex = new RegExp(req.authProvider.options.passwordRegex ?? '.*');
+  const test = passwordRegex.test(req.body.new);
```
Comment (gabila-r) [inline services/auth/lib/local.js:440]: Is it really necessary? Looks more like debugging, doesn't it?

File: services/docs/lib/services/document.js [hunk 2]
```diff
@@ -169,8 +169,7 @@ module.exports.lockDocument = async function (resource, state, filter, tokenTime
       }
     };
 
-    const result = await lockCollection.findOneAndUpdate(find, update);
-    return result;
+    return await lockCollection.findOneAndUpdate(find, update);
```
Comment (gabila-r) [inline services/docs/lib/services/document.js:172]: Here the return value changes 🤔 Is it not a problem?

**Review by gabila-r — APPROVED** [review]
Nice job ! 👏 Just 2 little warning 🙏

File: services/auth/lib/local.js (hunk 1 above)
Comment (rombat) [reply services/auth/lib/local.js:440]: yep you're right

File: services/docs/lib/services/document.js (hunk 2 above)
Comment (rombat) [reply services/docs/lib/services/document.js:172]: the only place where `lockDocument()` is used is [here](https://github.com/campsi/campsi-mono/blob/176de90cfd047805afc5f4565e59026f090ccd61/services/docs/lib/handlers.js#L288), and it just returns the result with the response, so it's ok

**Pull request merged**